routing/payment_lifecycle.go

```diff
@@ -187,6 +187,11 @@ lifecycle:
 			// the preimage and route.
 			for _, a := range payment.HTLCs {
 				if a.Settle != nil {
+					err := p.router.cfg.Control.DeleteFailedAttempts(p.identifier)
```

Comment:
Given that we have a settled payment here, isn't that a problem that we do fail the lifecycle if the delete fails for whatever reason? Perhaps we could log and error or critical instead? @yyforyongyu

Reply:
Very good point! Yeah we dare not fail the payment here since it's settled. Just to be safe we should log an error here instead.

Reply:
Ahh good catch, that makes sense! Done.

Reply:
hmmm did you forget to push the latest update?